### PR TITLE
CrossCor layer

### PR DESCRIPTION
**Pull request by ayush-1506**
Same as #423 (which could be edited since I lost access to that github account).

### CODE REVIEW

File: src/layers/conv.jl [hunk 1]
```diff
@@ -199,6 +199,66 @@ end
 (a::DepthwiseConv{<:Any,<:Any,W})(x::AbstractArray{<:Real}) where {T <: Union{Float32,Float64}, W <: AbstractArray{T}} =
   a(T.(x))
 
+  """
```
Comment (MikeInnes) [inline src/layers/conv.jl]: Formatting is off here

File: test/cuda/cuda.jl [hunk 2]
```diff
@@ -36,6 +36,10 @@ c = gpu(Conv((2,2),3=>4))
 l = c(gpu(rand(10,10,3,2)))
 Flux.back!(sum(l))
 
+c = gpu(CrossCor((2,2),3=>4))
+l = c(gpu(rand(10,10,3,2)))
+Flux.back!(sum(l))
```
Comment (MikeInnes) [inline test/cuda/cuda.jl]: Would be nice to have some numerical tests here, e.g. comparing to a regular convolution. Also, let's have a news item for this.

**Comment by dhairyagandhi96** [comment]
Not to get ahead of myself, we'll need a News item.

**Comment by dhairyagandhi96** [comment]
bors try

**Comment by dhairyagandhi96** [comment]
Could you do a quick conflict resolve?

**Comment by ayush-1506** [comment]
Done!

**Comment by dhairyagandhi96** [comment]
bors try

**Comment by MikeInnes** [comment]
Can we also link `CrossCor` in the layer reference?

@avik-pal once the docs are updated, if you're happy with the tests here go ahead and merge with bors.

**Comment by MikeInnes** [comment]
Thanks @ayush-1506!

**Pull request merged**